cmd/main.go

```diff
@@ -183,6 +193,22 @@ func main() {
 		log.Fatalf("not supported deploy target %s", Options.DeployTarget)
 	}
 
+	hostApi := host.NewManager(log.WithField("pkg", "host-state"), db, eventsHandler, hwValidator,
+		instructionApi, &Options.HWValidatorConfig, metricsManager, &Options.HostConfig, lead)
+	clusterApi := cluster.NewManager(Options.ClusterConfig, log.WithField("pkg", "cluster-state"), db,
+		eventsHandler, hostApi, metricsManager, lead)
+	expirer := imgexpirer.NewManager(objectHandler, eventsHandler, Options.BMConfig.ImageExpirationTime, lead)
+
+	// Start monitors threads
+	monitors := []thread.PeriodicTask{
+		{Name: "Cluster State Monitor", Interval: Options.ClusterStateMonitorInterval, Exec: clusterApi.ClusterMonitoring},
+		{Name: "Host State Monitor", Interval: Options.HostStateMonitorInterval, Exec: hostApi.HostMonitoring},
+		{Name: "Image Expiration Monitor", Interval: Options.ImageExpirationInterval, Exec: expirer.ExpirationTask}}
+
+	monitorsRunner := thread.NewPeriodicTasksRunner(log.WithField("pkg", "cluster-monitor"), monitors)
```

Comment:
I don't think that it is needed but it's up to you

Reply:
i like it :)